src/WalkingModule/src/Module.cpp

```diff
@@ -693,12 +695,22 @@ bool WalkingModule::updateModule()
             return false;
         }
 
-        if (!evaluateZMP(measuredZMP))
+        // compute the global CoP
+        if (!computeGlobalCoP(iDynTree::toEigen(measuredZMP)))
         {
-            yError() << "[WalkingModule::updateModule] Unable to evaluate the ZMP.";
+            yError() << "[WalkingModule::updateModule] Unable to compute the global CoP.";
             return false;
         }
 
+        // remove the ZMP offset if required
+        if (m_removeZMPOffset)
+        {
+            // remove rotated offset
+            iDynTree::Rotation yawRotation = this->computeAverageYawRotationFromPlannedFeet();
+            m_zmpOffset = yawRotation * m_zmpOffsetLocal;
+            iDynTree::toEigen(measuredZMP) += iDynTree::toEigen(m_zmpOffset).head<2>();
+        }
```

Comment:
Isn't also this supposed to go in the function?

Reply:
This is actually this required only in the update module since in start walking we need to compute it

Reply:
Ah I see, sounds good then!